pootle/apps/pootle_store/models.py

```diff
@@ -2272,21 +2276,35 @@ def update_store_header(self, user=None):
             mtime = self.get_cached_value(CachedMethods.MTIME)
             if mtime is None or mtime == datetime_min:
                 mtime = timezone.now()
+            user_displayname = None
+            user_email = None
             if user is None:
+                submissions = self.translation_project.submission_set.exclude(
+                    submitter__username="nobody")
                 try:
-                    submit = self.translation_project.submission_set \
-                                 .filter(creation_time=mtime).latest()
-                    if submit.submitter.username != 'nobody':
-                        user = submit.submitter
-                except ObjectDoesNotExist:
+                    username, fullname, user_email = (
+                        submissions.filter(creation_time=mtime).values_list(
+                            "submitter__username",
+                            "submitter__full_name",
+                            "submitter__email").latest())
+                except Submission.DoesNotExist:
                     try:
-                        lastsubmit = self.translation_project.submission_set \
-                                                             .latest()
-                        if lastsubmit.submitter.username != 'nobody':
-                            user = lastsubmit.submitter
-                        mtime = min(lastsubmit.creation_time, mtime)
+                        mtime, username, fullname, user_email = (
+                            submissions.values_list(
+                                "creation_time",
+                                "submitter__username",
+                                "submitter__full_name",
+                                "submitter__email").latest())
                     except ObjectDoesNotExist:
                         pass
+                if user_email:
+                    user_displayname = (
+                        fullname.strip()
+                        if fullname.strip()
+                        else username)
+            elif user.is_authenticated:
```

Comment:
As commented by @julen https://github.com/translate/pootle/pull/4387#issuecomment-172566849 the `user` passed to `store.sync` is dead code, so the `user` passed to `update_store_header` is dead code too, and this branch is never run.